### ISSUE TITLE
Basicauth documentation and example issues

### ISSUE THREAD

**Issue by a-a-abramov** [spec-first/connexion#1485]
### Description
`examples/openapi3/basicauth` example is broken right now. I assume it was broken since #869 (commit 8830d56841d29fe2ddce9d8507d60a27594cff30) - the example tries to import `connexion.decorators.security` which is now deleted. There's also almost the same issue in the docs [here](https://connexion.readthedocs.io/en/latest/security.html#oauth-2-authentication-and-authorization) - `connexion.decorators.security.validate_scope` function is obviously deleted with the whole file. 

It's very easy to fix (I will create a PR once any feedback is given), but unfortunately this example has code that makes no sence - it sets and checks scopes, while Swagger/OpenAPI 3 docs clearly state, that this method does not support scoping. Your [basic auth documentation](https://connexion.readthedocs.io/en/latest/security.html#basic-authentication) implicitly says the same:

> If the security declaration of the operation also has an oauth security requirement, required_scopes is taken from there, otherwise it’s None. This allows authorizing individual operations with [oauth scope](https://oauth.net/2/scope/) while using basic authentication for authentication.

As I understand this you can define multiple security schemes and use them in conjunction (AND). And then authn from **basic** and authz with scopes from **oauth**. But I don't really see the point of this, because you actually have to perform authn twice - I found no way to bypass/ignore oauth authn in this case. You may implement service account approach, but basic authn in this case becomes obsolete. 

It would be awesome if you could provide example of basicauth + oauth in case I'm completely wrong! Or otherwise just remove this clause completely and state that it's impossible to use scopes unless oauth2 is used. I, once again, can fix this section of docs and create another PR.

### Expected behaviour
All examples work.
Documentation is unambiguous and clear for new users like myself.

### Actual behaviour
`openapi3/basicauth` example fails due to previous refactoring.
Documentation is lacking clarity or additional examples.

### Steps to reproduce
Run `openapi3/basicauth`

### Additional info:

```
$ python --version
Python 3.10.1
$ pip show connexion | grep "^Version\:"
Version: 2.12.0
```

**Comment by Ruwann** [comment]
Hi @a-a-abramov , thanks for the detailed report!

Indeed, as you say, the outdated documentation is already reported in #1471 .

Concerning the behaviour of making the scopes accessible to the basic auth security scheme when there is also an OAuth security requirement defined on the operation: I agree that is a bit of an abuse of the implementation as only OAuth and OpenID Connect support scopes on the operation. This is currently also being changed in #1474 as the current implementation also leads to errors in case there are multiple sets of scopes defined for an operation.

Feel free to open a PR for updating the documentation, I would gladly review it! You can definitely change the examples such that these scopes are also correctly used in the examples.

**Comment by a-a-abramov** [comment]
Hi @Ruwann !

Appreciate your response! You kinda proved I'm not that crazy as I thought I was. Simplification of basicauth example is here - #1486 

I will submit a draft of the documentation update a bit later.

**Comment by RobbeSneyders** [comment]
Thanks @a-a-abramov 
Closed by #1486
